docs/references/architecture.mdx

```diff
@@ -0,0 +1,103 @@
+
+Restate clusters are designed to scale out in support of large deployments. As you add more machines, it becomes wasteful to replicate all the functionality across all the machines in a cluster, since not all features need to scale out at the same rate. **Roles** control which features run on any given node, enabling specialization within the cluster.
+
+Here is an overview of the different roles that can run on a node:
+
+<img alt={"Restate Node internal architecture"} src={"/img/architecture/node_internal_architecture.png"}/>
+
+- Metadata server: the source of truth for cluster-wide information
+- Ingress: the entry point for external requests
+- Log server: responsible for durably persisting the log
+- Worker: houses the partition processors
+
+## Metadata server
+
+The Restate metadata store is part of the control plane and is the internal source of truth for node membership and responsibilities. It is essential to the correctness of the overall system: in a cluster this service enables distributed consensus about other components' configuration. All nodes in a Restate cluster must be able to access the metadata service, though not all members of the cluster need to be part of hosting it. Restate includes a built-in Raft-based metadata store which is hosted on all nodes running the `metadata-server` role.
+
+The metadata store is designed to support relatively low volumes of read and write operations (at least compared to other parts of Restate), with the highest level of integrity and availability.
+
+
+[//]: # (For some weird reason this shifts to the right if I use two hashtags)
+<h2>Ingress</h2>
+
+External requests enter the Restate cluster via the HTTP ingress component, which runs on nodes assigned the `http-ingress` role. Compared to other roles, the HTTP ingress role does not involve long-lived state and it can move around relatively freely, since it only handles ongoing client connections.
+
+<Admonition type="info" title={"Fine-grained ingress role"}>
+    The fine-grained `http-ingress` role is a new addition in Restate 1.2. For backwards-compatibility, nodes running the `worker` role will continue to run the ingress function in version 1.2.
+</Admonition>
+
+
+## Log servers
+
+Log server nodes running the `log-server` role are responsible for durably persisting the log. If the log is the equivalent of a WAL, then partition stores are the materializations that enable efficient reads of the events (invocation journals, key-value data) that have been recorded. Depending on the configured **log replication** requirements, Restate will allocate multiple log servers to persist a given log, and this will change over time to support maintenance and resizing of the cluster.
```

Comment:
```suggestion
Log server nodes running the `log-server` role are responsible for durably persisting the log. If the log is the equivalent of a WAL, then partition stores are the materializations that enable efficient reads of the events (invocation journals, key-value data) that have been recorded. Depending on the configured **log replication** requirements, Restate will replicate log records to multiple log servers to persist a given log, and this will change over time to support maintenance and resizing of the cluster.
```